src/Compilers/CSharp/Portable/Syntax/TypeDeclarationSyntax.cs

```diff
@@ -114,7 +115,8 @@ public static TypeDeclarationSyntax TypeDeclaration(SyntaxKind kind, string iden
                     return SyntaxFactory.InterfaceDeclaration(attributes, modifiers, keyword, identifier, typeParameterList, baseList, constraintClauses, openBraceToken, members, closeBraceToken, semicolonToken);
                 case SyntaxKind.RecordDeclaration:
                     return SyntaxFactory.RecordDeclaration(attributes, modifiers, keyword, classKeyword: default, identifier, typeParameterList, parameterList: null, baseList, constraintClauses, openBraceToken, members, closeBraceToken, semicolonToken);
-                // PROTOTYPE(record-structs): we likely need to handle RecordStructDeclaration
+                case SyntaxKind.RecordStructDeclaration:
+                    return SyntaxFactory.RecordStructDeclaration(attributes, modifiers, keyword, structKeyword: SyntaxFactory.Token(SyntaxKind.StructKeyword), identifier, typeParameterList, parameterList: null, baseList, constraintClauses, openBraceToken, members, closeBraceToken, semicolonToken);
                 default:
                     throw new ArgumentException("kind");
```

Comment:
not part of your change, so feel free to ignore, but it feels like we should use ExceptionUtilities.UnexpectedValue here.